src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './NewCommentForm.scss';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+import { NewComment } from '../../react-app-env';
+
+interface Props {
+  selectedPostId: number;
+  onAddComment: (newComment: NewComment) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPostId,
+  onAddComment,
+}) => {
+  const [newName, setNewName] = useState('');
+  const [newEmail, setNewEmail] = useState('');
+  const [newBody, setNewBody] = useState('');
+
+  const clearForm = () => {
+    setNewName('');
+    setNewEmail('');
+    setNewBody('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newComment: NewComment = {
+      postId: selectedPostId,
+      name: newName,
+      email: newEmail,
+      body: newBody,
+    };
+
+    onAddComment(newComment);
+    clearForm();
+  };
+
+  return (
+    <form className="NewCommentForm" onSubmit={handleSubmit}>
+      <div className="form-field">
+        <input
+          type="text"
+          name="name"
+          placeholder="Your name"
+          className="NewCommentForm__input"
+          required
+          value={newName}
+          onChange={event => setNewName(event.target.value)}
```

Comment:
unify onChange; when your state is an object, set it's props like so
```
const handleChange = useCallback( (event) => {
  const { name, value } = event.target;

  setInput(prevInput => ({
    ...prevInput,
    [name]: value,
  }))
}, [ ... ])
```

and use it across your inputs as unified method